packages/google_maps_flutter/google_maps_flutter_android/lib/src/google_maps_flutter_android.dart

```diff
@@ -796,6 +796,48 @@ class GoogleMapsFlutterAndroid extends GoogleMapsFlutterPlatform {
     return PlatformTileOverlay(
         json: tileOverlay.toJson() as Map<String, Object?>);
   }
+
+  static PlatformCameraUpdate _platformCameraUpdateFromCameraUpdate(
+      CameraUpdate update) {
+    if (update is NewCameraPosition) {
```

Comment:
You might have discussed this with @stuartmorgan if so feel free to resolve but can cameraUpdate include an enum of the type of update instead of these long object type checks?

Reply:
This was discussed a bit for the Pigeon classes (where I don't think the enum probably helps us), but not for the Dart code.

I think for Dart, where we will be creating a new permanent public API surface, it would be better to add a type enum, and to make all of these classes private, so that we're not locking ourselves into a specific class structure forever.

Reply:
If these classes were private, would that not then make their members inaccessible in this file?

Reply:
🤦🏻 

Sorry, I was thinking about mirroring the Java SDK's approach, but there they only need to consume the properties internally to the SDK.